src/component/Modal/Metadata/Metadata.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { Modal, Descriptions } from 'antd';
+
+import MetadataParser from '../../../util/MetadataParser';
+
+import './Metadata.less';
+
+const isEmpty = require('lodash/isEmpty');
+
+interface MetadataProps {
+  layer: any;
+  onCancel: (arg: any) => void;
+  t: (arg: string) => {};
+}
+
+interface MetadataState {
+  metadata: any;
+  error: string;
+}
+
+/**
+ * The Modal Metadata component
+ *
+ */
+export class Metadata extends React.Component<MetadataProps, MetadataState> {
+
+  /**
+   *
+   */
+  constructor(props: MetadataProps) {
+    super(props);
+
+    this.state = {
+      metadata: {},
+      error: ''
+    };
+  }
+
+  /**
+   * Called on lifecycle componentDidMount.
+   */
+  componentDidMount() {
+    this.fetchMetadata(this.props.layer.get('metadataIdentifier'));
+  }
+
+  /**
+   * Fetch metadata by uuid from GNOS
+   * @param layer
+   */
+  fetchMetadata(uuid: string) {
+    const url = '/globewq-webapp/metadata/getRecordByUuid.action?uuid=' + uuid + '&outputFormat=json';
```

Comment:
The above URL looks not very portable to my eyes. How can we have this more generic?

Reply:
whoops nice catch, yes this needs to be fixed

Reply:
fixed in https://github.com/terrestris/react-geo-baseclient/pull/654